modules/offers/client/components/OffersPresentational.js

```diff
@@ -0,0 +1,276 @@
+          <div className="icon-sofa icon-3x text-muted"></div>
+
+          {/*  Show for others  */}
+          {!isOwnOffer &&
+            <h4>
+              {t('Sorry, user is not hosting currently.')}
+            </h4>
+          }
+
+          {/*  Show for the user  */}
+          {isOwnOffer &&
+            <div>
+              <br />
+              <p className="lead">
+                <em>{t('Offering hospitality and welcoming “strangers” to our homes strengthens our faith in each other.')}</em>
+              </p>
+              <br />
+            </div>
+          }
+        </div>
+        }
+
+        {/*  Start hosting and meet people action buttons  */}
+        {isOwnOffer && (!offer.status || offer.status === 'no') &&
+        <div className="text-center">
+          <br />
+          <hr className="hr-gray hr-tight hr-xs" />
+          <a href={'/offer/host?status=yes'}className="btn btn-inverse-primary">
+            Start hosting travellers
+          </a>
+          &nbsp;
+          <a href={'/offer/meet'} className="btn btn-inverse-primary">
+            Meet people
+          </a>
+        </div>
+        }
+      </div>
+    );
+  }
+
+  renderMap() {
+    const { isMobile } = this.state;
+    const { offer, t } = this.props;
+    return (
+      <div>
+        {(offer.status === 'yes' || offer.status === 'maybe') &&
+          <OfferLocation offer={offer} ></OfferLocation>
+        }
+        {(offer.status === 'yes' || offer.status === 'maybe') &&
+        <div className="panel-footer text-center">
+          <a href={`/search?offer=${offer._id}`} className="btn btn-sm btn-inverse-primary">
+            {t('Bigger map')}
+          </a>
+          {isMobile &&
+          <a href="geo:{{offer.location[0]}},{{offer.location[1]}};u=200"
+            className="btn btn-sm btn-inverse-primary">
+            {t('Open on device')}
+          </a>
+          }
+        </div>
+        }
+      </div>
+    );
+  }
+
+  renderOffer() {
+    const { t } = this.props;
+    const { isOwnOffer, offer } = this.props;
+    return (
+      <div className="panel panel-default offer-view">
+        <div className="panel-heading">
+          {t('Accommodation')}
+          {/*  Button + dropdown for user's own profile  */}
+          {isOwnOffer && this.renderButtonOwn()}
+          {/*  Button for other profiles  */}
+          {!isOwnOffer && this.renderButtonOther()}
+        </div>
+
+        {/*  Show offer  */}
+        <div className="panel-body">
+          {/*  Hosting: yes | maybe  */}
+          {(offer.status && offer.status !== 'no') && this.renderHostingYesMaybe()}
+
+          {/*  Hosting: no  */}
+          {(!offer || !offer.status || offer.status === 'no') && this.renderHostingNo()}
+        </div>
+
+        {/*  The map (React component)  */}
+        {this.renderMap()}
+      </div>
+    );
+  };
+
+  render(){
+    return (this.props.isOwnOffer || this.props.isUserPublic) &&
```

Comment:
We'll need to ensure return `null` when there's nothing to render, so something like:
```js
    return this.props.isOwnOffer || this.props.isUserPublic
      ? this.renderOffer()
      : null;
```

Reply:
I didn't know about returning `null`, and just to add a bit more detail:
- if you are doing conditional rendering inside some jsx, false(y) is still fine (see https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)
- returning null from a render function causes the whole component not to render at all as if the parent didn't use it (see https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering) - it says this is for _rare cases_. I imagine in normal cases the parent component is in charge of deciding whether to render at all, and the child one would normally render _something_ (which might just be an error/warning message if it has invalid props/state/whatever).

Reply:
Right on! Not entirely sure how/with what I mixed this up. :-)